components/in_memory_engine/src/config.rs

```diff
@@ -130,7 +133,11 @@ impl InMemoryEngineConfig {
             );
         }
 
-        if self.evict_threshold.is_none() {
+        // Set default values for evict_threshold and stop_load_threshold if
+        // they are not set or the configuration is changed online.
+        if self.evict_threshold.is_none()
+            || online_changes.is_some_and(|c| c.contains_key("capacity"))
```

Comment:
How about check whether the incoming `ConfigChange` contains `capacity` in advance ?
Such as:
```Rust
let change_capacity = online_changes.is_some_and(|c| c.contains_key("capacity"));
if self.evict_threshold.is_none() || change_capacity {
   ...
}

```